cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sources/JmsSource.java

```diff
@@ -47,14 +53,21 @@
  * JMS Consumer and send the message as String to the CDAP ETL Template flow via {@link Emitter}
  * </p>
  */
-public class JmsSource extends RealtimeSource<String> {
+@Plugin(type = "source")
+@Name("JMS")
+@Description("JMS Realtime Source")
+public class JmsSource extends RealtimeSource<StructuredRecord> {
   private static final Logger LOG = LoggerFactory.getLogger(JmsSource.class);
 
+  public static final String JAVA_NAMING_PREFIX = "java.naming";
   public static final String JMS_DESTINATION_NAME = "jms.destination.name";
+  public static final String JMS_MESSAGES_TO_RECEIVE = "jms.messages.receive";
 
   private static final long JMS_CONSUMER_TIMEOUT_MS = 2000;
-  private static final String CDAP_JMS_SOURCE_NAME = JmsSource.class.getSimpleName();
-  private static final String JMS_MESSAGES_TO_RECEIVE = "jms.messages.receive";
+
+  public static final String MESSAGE = "message";
+
+  private final JmsConfig config;
```

Comment:
How is the config object used in the code? Shouldn't the properties be accessed through this class?